programming_examples/basic/passthrough_kernel/README.md

```diff
@@ -0,0 +1,100 @@
+
+It is important to note that the Shim Tile and Compute Tile DMAs move data concurrently, and the Compute Tile's AIE Core is also processing data concurrent with the data movement. This is made possible by expressing depth `2` in declaring, for example, `object_fifo("in", ShimTile, ComputeTile2, 2, memRef_ty)` to denote ping-pong buffers.
+
+## Design Component Details
+
+### AIE Array Structural Design
+
+This design performs a memcpy operation on a vector of input data. The AIE design is described in a python module as follows:
+
+1. **Constants & Configuration:** The script defines input/output dimensions (`N`, `n`), buffer sizes in `lineWidthInBytes` and `lineWidthInInt32s`, and tracing support.
+
+1. **AIE Device Definition:** `@device` defines the target device. The `device_body` function contains the AIE array design definition.
+
+1. **Scaling Function Declarations:** `passThroughLine` is an external function imported from `passThrough.cc`.
+
+1. **Tile Definitions:** `ShimTile` handles data movement, and `ComputeTile2` processes the scaling operations.
+
+1. **Object Fifos:** `of_in` and `of_out` are defined to facilitate communication between `ShimTile` and `ComputeTile2`.
+
+1. **Tracing Flow Setup (Optional):** A circuit-switched flow is set up for tracing information when enabled.
+
+1. **Core Definition:** The `core_body` function loops through sub-vectors of the input data, acquiring elements from `of_in`, processing using `passThroughLine`, and outputting the result to `of_out`.
+
+1. **Data Movement Configuration:** The `sequence` function configures data movement and synchronization on the `ShimTile` for input and output buffer management.
+
+1. **Tracing Configuration (Optional):** Trace control, event groups, and buffer descriptors are set up in the `sequence` function when tracing is enabled.
+
+1. **Generate the design:** The `passthroughKernel()` function triggers the code generation process. The final print statement outputs the MLIR representation of the AIE array configuration.
+
+### AIE Core Kernel Code
+
+`passThrough.cc` contains a C++ implementation of vectorized memcpy operation designed for AIE cores. It consists of two main sections:
+
+1. **Vectorized Copying:** The `passThrough_aie()` function processes multiple data elements simultaneously, taking advantage of AIE vector datapath capabilities to load, copy and store data elements.
+
+1. **C-style Wrapper Functions:** `passThroughLine()` and `passThroughTile()` are two C-style wrapper functions to call the templated `passThrough_aie()` vectorized memcpy implementation from the AIE design implemented in `aie2.py`. The `passThroughLine()` and `passThroughTile()` functions are compiled for `uint8_t`, `int16_t`, or `int32_t` determined by the value the `BIT_WIDTH` variable defines. 
+
+## Usage
+
+### C++ Testbench
+
+To compile the design and C++ testbench:
+
+```
+make
+make build/passThroughKernel.exe
+```
+
+To run the design:
+
+```
+make run
```

Comment:
only make run needed

Reply:
I'm going to leave it, even though it is pedantic